src/GridFieldNestedForm.php

```diff
@@ -0,0 +1,517 @@
+                return '';
+            }
+            if ($this->canExpandCheck) {
+                if (is_callable($this->canExpandCheck)
+                    && !call_user_func($this->canExpandCheck, $record)
+                ) {
+                    return '';
+                } elseif (is_string($this->canExpandCheck)
+                    && $record->hasMethod($this->canExpandCheck)
+                    && !$record->{$this->canExpandCheck}($record)
+                ) {
+                    return '';
+                }
+            }
+            $toggle = 'closed';
+            $className = str_replace('\\', '-', get_class($record));
+            $state = $gridField->State->GridFieldNestedForm;
+            $stateRelation = $className.'-'.$record->ID.'-'.$this->relationName;
+            $openState = $state && (int)$state->getData($stateRelation) === 1;
+            $forceExpand = $this->expandNested && $record->$relationName()->count() > 0;
+            if (!$this->forceCloseNested
+                && ($forceExpand || $openState)
+            ) {
+                $toggle = 'open';
+            }
+
+            GridFieldExtensions::include_requirements();
+
+            return ViewableData::create()->customise([
+                'Toggle' => $toggle,
+                'Link' => $this->Link($record->ID),
+                'ToggleLink' => $this->ToggleLink($record->ID),
+                'PjaxFragment' => $stateRelation,
+                'NestedField' => ($toggle == 'open') ? $this->handleNestedItem($gridField, null, $record): ' '
+            ])->renderWith('Symbiote\GridFieldExtensions\GridFieldNestedForm');
+        }
+    }
+    
+    public function getURLHandlers($gridField)
+    {
+        return [
+            'nested/$RecordID/$NestedAction' => 'handleNestedItem',
+            'toggle/$RecordID' => 'toggleNestedItem',
+            'POST movetoparent' => 'handleMoveToParent'
+        ];
+    }
+
+    /**
+     * @param GridField $gridField
+     * @return array
+     */
+    public function getHTMLFragments($gridField)
+    {
+        /**
+         * If we have a DataObject with the hierarchy extension, we want to allow moving items to a new parent.
+         * This is enabled by setting the data-url-movetoparent attribute on the grid field, so that the client
+         * javascript can handle the move.
+         * Implemented in getHTMLFragments since this attribute needs to be added before any rendering happens.
+         */
+        if (is_a($gridField->getModelClass(), DataObject::class, true)
+            && DataObject::has_extension($gridField->getModelClass(), Hierarchy::class)
+        ) {
+            $gridField->setAttribute('data-url-movetoparent', $gridField->Link('movetoparent'));
+        }
+        return [];
+    }
+
+    /**
+     * Handle moving a record to a new parent
+     *
+     * @return string
+     */
+    public function handleMoveToParent(GridField $gridField, $request)
```

Comment:
```suggestion
     */
    public function handleMoveToParent(GridField $gridField, $request): string
```
For all methods which are not inherited from a parent class or interface, please use strict typehints instead of phpdocs, e.g. handleNestedItem. I won't list them all.